### PR TITLE
feat: support escaped paths in field name

### PR DESCRIPTION
**Pull request by chrisirhc**
Attempts to address https://github.com/TanStack/form/discussions/532 . 

This is a sketch implementation for feedback, using lodash's internals for get. 
Happy to roll a custom implementation but I thought of using an existing battle-tested implementation. 

This implementation handles some of the cases in https://github.com/lodash/lodash/blob/main/test/toPath.spec.js .

### CODE REVIEW

File: packages/form-core/src/utils.ts [hunk 1]
```diff
@@ -110,35 +110,49 @@ export function deleteBy(obj: any, _path: any) {
...
+    if (quote) {
+      key = subString.replace(reEscapeChar, '$1')
+    }
+    else if (expression) {
+      key = expression.trim()
+    }
+    // key difference from lodash string to path algoritm
+    if ((/^\d+$/).test(key)) {
+      key = parseInt(key, 10);
+    }
```
Comment (chrisirhc) [inline packages/form-core/src/utils.ts:151]: Main modification I made based on the previous logic in `makePathArray`.

**Comment by crutchcorn** [comment]
This looks good overall! The only concern I have now is typings. We'll want a way to merge the types from `string[]` to `string`. This can be done like so:

https://github.com/crutchcorn/ts-util-helpers/blob/main/src/concat-string.ts#L1-L7

But then the question becomes "how do we get `TName` inference working for this?

**Comment by chrisirhc** [comment]
> This looks good overall! The only concern I have now is typings. We'll want a way to merge the types from `string[]` to `string`. 
Not sure I get this part of your comment. Where are you referring to the `string[]` type? Is it the `makePathArray`?

> But then the question becomes "how do we get `TName` inference working for this?
Just to make sure I get what you're referring to. You're referring to making sure that the `DeepKeys`  https://github.com/TanStack/form/blob/77f119b0a6e6caccb8fdc3d3cdeda97b3560d1e5/packages/form-core/src/utils.ts#L299 can work for the `['']` syntax, right? I'll look in this.

**Comment by crutchcorn** [comment]
I wouldn't modify `DeepKeys`. I would modify the instance where `TName` is being passed and introduce a new utility type of some kind.

**Comment by chrisirhc** [comment]
Added spec updates and also utilities as discussed. Ready for comment/review.

**Comment by crutchcorn** [comment]
Hey, I am super sorry - but we've made **drastic** changes to the internals of these files. It's honestly better to close out this PR and revisit them. I _think_ our types support this now, but our logic certainly doesn't.

Let's track the progress of this in: #639

**Pull request closed without merging**